openapiart/openapiartgo.py

```diff
@@ -1063,6 +1063,19 @@ def _write_interface(self, new):
                 }}
                 return str
             }}
+
+            func (obj *{struct}) Clone() ({interface}, error) {{
+                newObj := New{interface}()
+                pbText, err := obj.ToPbText()
```

Comment:
I would recommend marshelling to proto bytes instead of proto text (since it will be more efficient).

https://gist.github.com/miguelmota/25568433ad8cfddb5ea556a5644c9fde#file-main-go-L16